### ISSUE TITLE
`buildLocation` does not encode slashes in path params

### ISSUE THREAD

**Issue by andrejilderda** [TanStack/router#1515]
### Describe the bug

When using buildLocation() with params, I expect the values of the params to be encoded. This is the case for spaces and other special characters, but not for slashes. This results in invalid router url's.

### Your Example Website or App

https://stackblitz.com/edit/github-lxalsq?file=src%2Froutes%2Findex.tsx

### Steps to Reproduce the Bug or Issue

__Also see the StackBlitz example above.__

1. Use `buildLocation` with a dynamic route segment and pass a string with a slash.
```
  router.buildLocation({
    from: '<routeWithDynamicSegment>', 
    params: { dynamicSegment: 'foo bar/baz' }
  })
```
2. Notice how the generated `.href` is encoded, except for the slash: `/foo%20bar/baz`.
3. The naive solution would be to apply the encoding to the parameters passed to buildLocation():
```
...
  params: { dynamicSegment: encodeURIComponent(stringWithSlash) }}) };
...
```
4. But this results in the encoding of the encoded characters…:
`/foo%2520bar%252Fbaz` 

### Expected behavior

I expected the slash to be encoded for every param passed to `buildLocation`. Going further on the example above I expected the output to be:
`/foo%20bar%2Fbaz`

### Platform

- Not applicable.

### Additional context

Thanks for all your work! 🫶️

**Comment by SeanCassiere** [comment]
Released the fix for this in the [1.30.1 release](https://github.com/TanStack/router/releases/tag/v1.30.1).

**Comment by andrejilderda** [comment]
Wow, thanks for the super fast fix @SeanCassiere!! 🥇
